# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/cairo/cairo/src/win32/cairo-win32-gdi-compositor.c
# Commit: 3cdfb39376b7
# Full Hash: 3cdfb39376b736597fe658a322389117686b7e82
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2024-05-07 09:13:07
# Regressor Bug: 1892913
# File Overlap Count: 3
# Description:
#   Bug 1892913 - patch 1 - Update cairo source to 1.18.0 r=gfx-reviewers,lsalzman
#   
#   Wholesale replacement of files in gfx/cairo/cairo with contents of the 1.18.0 release tarball,
#   omitting subdirectories that are not part of the core library build.
#   
# ==============================================================================

diff -r 9daa351f875b -r 3cdfb39376b7 gfx/cairo/cairo/src/win32/cairo-win32-gdi-compositor.c
--- a/gfx/cairo/cairo/src/win32/cairo-win32-gdi-compositor.c	Mon May 06 19:22:35 2024 +0000
+++ b/gfx/cairo/cairo/src/win32/cairo-win32-gdi-compositor.c	Mon May 06 19:25:12 2024 +0000
@@ -602,7 +602,8 @@
     if (! _cairo_clip_is_region (composite->clip))
 	return FALSE;
 
-    if (cairo_scaled_font_get_type (scaled_font) != CAIRO_FONT_TYPE_WIN32)
+    cairo_font_type_t type = cairo_scaled_font_get_type (scaled_font);
+    if (type != CAIRO_FONT_TYPE_WIN32 && type != CAIRO_FONT_TYPE_DWRITE)
 	return FALSE;
 
     if (! _cairo_pattern_is_opaque_solid (&composite->source_pattern.base))